tools/run-mainnet.sh

```diff
@@ -1,22 +1,39 @@
-NAMESPACE=main
+#!/bin/bash
+# Copyright (c) 2022 The MobileCoin Foundation
+NAMESPACE=prod
+NET=main
 
-WORK_DIR="$HOME/.mobilecoin/${NAMESPACE}"
+WORK_DIR="$HOME/.mobilecoin/${NET}"
 WALLET_DB_DIR="${WORK_DIR}/wallet-db"
 LEDGER_DB_DIR="${WORK_DIR}/ledger-db"
-mkdir -p ${WORK_DIR}
 
-(cd ${WORK_DIR} && CONSENSUS_SIGSTRUCT_URI=$(curl -s https://enclave-distribution.${NAMESPACE}.mobilecoin.com/production.json | grep consensus-enclave.css | awk '{print $2}' | tr -d \" | tr -d ,)
-curl -O https://enclave-distribution.${NAMESPACE}.mobilecoin.com/${CONSENSUS_SIGSTRUCT_URI})
+# Pass "--no-build" if the user just wants to run what they have in  
+# .mobilecoin/main instead of building and copying over a new exectuable
+if [ $# -eq 0 ] || [ $1 != "--no-build" ]; then
+    echo "Building"
+    mkdir -p ${WORK_DIR}
+    SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
-(cd ${WORK_DIR} && INGEST_SIGSTRUCT_URI=$(curl -s https://enclave-distribution.${NAMESPACE}.mobilecoin.com/production.json | grep ingest-enclave.css | awk '{print $2}' | tr -d \" | tr -d ,)
-curl -O https://enclave-distribution.${NAMESPACE}.mobilecoin.com/${INGEST_SIGSTRUCT_URI})
+    (cd ${WORK_DIR} && INGEST_SIGSTRUCT_URI=$(curl -s https://enclave-distribution.${NAMESPACE}.mobilecoin.com/production.json | grep ingest-enclave.css | awk '{print $2}' | tr -d \" | tr -d ,)
+    curl -O https://enclave-distribution.${NAMESPACE}.mobilecoin.com/${INGEST_SIGSTRUCT_URI})
+    INGEST_ENCLAVE_CSS="$WORK_DIR/ingest-enclave.css"
```

Comment:
This doesn't seem related to building full-service, it is related to running it.
Any reason not to always try and download this file if it doesn't exist in the work directory?
That would also remove the need to abort if the user didn't explicitly set an environment variable with it.

Reply:
I don't think we should overwrite files that already exist, but no reason not to download it if it doesn't exist and we know where to get it from